src/components/screen.cpp

```diff
@@ -107,7 +108,12 @@ void screen::handle(const evt::randr_screen_change_notify& evt) {
   if (screen->width_in_pixels != m_size.w || screen->height_in_pixels != m_size.h) {
     changed = true;
   } else {
-    changed = have_monitors_changed();
+    auto p = have_monitors_changed();
+    changed = p.first;
+    if (p.second == 0) {
```

Comment:
Might be worth moving this one level up, outside of the `else` block so that we also don't reload if the screen size changed *and* there are no monitors.